app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -150,25 +217,28 @@ def save_container_routes_inventory(ems, hashes, target = nil)
     store_ids_for_new_records(ems.container_routes, hashes, :ems_ref)
   end
 
-  def save_container_nodes_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_nodes.reset
-    deletes = if target.kind_of?(ExtManagementSystem)
-                :use_association
-              else
-                []
-              end
-
-    save_inventory_multi(ems.container_nodes, hashes, deletes, [:ems_ref],
-                         [:labels, :tags, :computer_system, :container_conditions,
-                          :additional_attributes], [:namespace])
-    store_ids_for_new_records(ems.container_nodes, hashes, :ems_ref)
+  def graph_container_nodes_inventory(ems, hashes, target = nil)
+    hashes.to_a.each do |h|
+      h = h.except(:labels, :tags, :additional_attributes) # TODO children
+      h = h.except(:namespace)
+      node = @inv_collections[:container_nodes].lazy_find(h[:ems_ref])
+      graph_container_node_conditions_inventory(node, h.delete(:container_conditions))
+      graph_computer_system_inventory(node, h.delete(:computer_system))
+      @inv_collections[:container_nodes].build(h)
+    end
   end
 
-  def save_computer_system_inventory(container_node, hash, _target = nil)
-    save_inventory_single(:computer_system, container_node, hash, [:hardware, :operating_system])
+  def graph_computer_system_inventory(parent, hash, _target = nil)
+    return if hash.nil?
+    hash = hash.merge(:managed_entity => parent)
+    # TODO: there is probably is shorter way to link them?
```

Comment:
Right, so again, take IC as a model of a DB table. So the IC should contain the relations that will become the foreign keys later.

In your example, 'a_id' is a foreign_key of 'b' pointing to 'a', so it will be saved as part of 'b' table.

At some point we could extend the API to be more Railsy and define it on 'a' side, while it would seamlessly move to 'b'. Well the 'a' side has has many, so you would have to actually distribute it to many InventoryObjects of the 'b'. So it's doable, but for now the API is simple, you define a flat list of attributes, that has a correlation to the actual DB table.